core/src/test/scala/unit/kafka/integration/AlterIsrRequestTest.scala

```diff
@@ -0,0 +1,117 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package unit.kafka.integration
+
+import kafka.server.KafkaConfig
+import kafka.utils.TestUtils
+import kafka.zk.ZooKeeperTestHarness
+import org.apache.kafka.clients.admin.{Admin, MoveControllerOptions}
+import org.apache.kafka.clients.producer.{Producer, ProducerRecord}
+import org.apache.kafka.common.{Node, TopicPartition}
+import org.junit.jupiter.api.Assertions.{assertEquals, assertTrue}
+import org.junit.jupiter.api.Test
+
+import java.nio.charset.StandardCharsets
+import java.util
+import java.util.{Collections, Properties}
+
+class AlterIsrRequestTest extends ZooKeeperTestHarness {
+  @Test
+  def testUnauthorizedAlterISRRequest(): Unit = {
+    // create brokers
+    val totalBrokers = 4
+    val serverConfigs = TestUtils.createBrokerConfigs(totalBrokers, zkConnect, false)
+      .map(props => {
+        if (props.get(KafkaConfig.BrokerIdProp).equals((totalBrokers - 1).toString)) {
+          props.setProperty(KafkaConfig.LiDenyAlterIsrProp, "true")
```

Comment:
Do we really need to add a config just for one test? Is there any workaround for this test? e.g., adding the broker to the controller's blacklist to block the request from broker to the first controller, adding some mock to the first controller to drop request?

Reply:
Per the discussion offline, adding this config is an easy way to reproduce the problem seen in prod.

I tried constructing a mock object to reproduce the problem, but it would also require very intrusive changes to the implementation to inject the mock object. In comparison, I like this customized config better.